firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/FirebaseAppDistribution.java

```diff
@@ -129,12 +133,16 @@ public UpdateTask updateIfNewReleaseAvailable() {
       if (cachedUpdateIfNewReleaseTask != null && !cachedUpdateIfNewReleaseTask.isComplete()) {
         return cachedUpdateIfNewReleaseTask;
       }
-
       cachedUpdateIfNewReleaseTask = new UpdateTaskImpl();
     }
-    checkForNewRelease()
-        .onSuccessTask(
-            release -> {
+
+    showSignInDialog()
+        .onSuccessTask(unused -> signInTester())
+        .onSuccessTask(unused -> checkForNewRelease())
+        .continueWithTask(
```

Comment:
If you're just calling getResult() and allowing the errors to propagate, then AFAICT you may as well use `onSuccessTask()` here.